src/plugins/codeStyle/index.ts

```diff
@@ -35,6 +35,21 @@ export default class CodeStyle {
         const requireConditionGroup = conditionStyle === 'group';
         const validateAAStyle = aaCommaStyle !== 'off';
         const walkExpressions = validateAAStyle;
+        const validateNewlineLast = newlineLast !== 'off';
+        const disallowNewlineLast = newlineLast === 'never';
+
+        if (validateNewlineLast &&
+            // Do not validate empty files
+            file.parser.tokens.filter(t => t.kind !== TokenKind.Newline && t.kind !== TokenKind.Eof).length > 0) {
```

Comment:
`tokens.filter` will process the entire file's tokens (creating a new array with it) and seems like a huge waste.
I'd recommend a better logic walking from the end.

Reply:
I don't see the automatic fix so I guess it's WIP - please mark the PR as DRAFT in this case.

Reply:
Fixed by 95a4f6e